demo/material/text-field.html

```diff
@@ -0,0 +1,129 @@
+      opacity: 0;
+    }
+  </style>
+</head>
+
+<body unresolved>
+
+  <div class="vertical-section-container centered">
+
+    <demo-navigation>
+      <h3 slot="header">Material theme</h3>
+    </demo-navigation>
+
+    <h1>Text field</h1>
+
+    <h3>Label and placeholder</h3>
+    <demo-snippet>
+      <template>
+        <vaadin-text-field placeholder="Placeholder"></vaadin-text-field>
+
+        <vaadin-text-field label="Label"></vaadin-text-field>
+
+        <vaadin-text-field label="Label and placeholder" placeholder="Placeholder"></vaadin-text-field>
+
+        <vaadin-text-field label="Label always floating" theme="always-float-label"></vaadin-text-field>
+
+      </template>
+    </demo-snippet>
+
+    <h3>Prefix and suffix</h3>
+    <demo-snippet>
+      <template>
+        <vaadin-text-field class="email" label="Email address" placeholder="Firstname" theme="always-float-label">
+          <div class="letter-icon" slot="prefix">✉️</div>
+          <div class="domain" slot="suffix">@vaadin.com</div>
+        </vaadin-text-field>
+
+        <!-- A bit of styling for the prefix and suffix elements -->
+        <custom-style>
+          <style>
+            vaadin-text-field.email {
+              width: 250px;
+            }
+
+            .letter-icon {
+              font-size: 24px;
+              padding-right: 5px;
+              align-items: center;
+            }
+
+            .domain {
+              font-size: 16px;
+            }
+          </style>
+        </custom-style>
+      </template>
+    </demo-snippet>
+
+    <h3>States</h3>
+    <demo-snippet>
+      <template>
+        <vaadin-text-field label="Focused" value="Value" focused></vaadin-text-field>
+
+        <vaadin-text-field label="Disabled" value="Value" disabled></vaadin-text-field>
+
+        <vaadin-text-field label="Read-only" value="Value" readonly></vaadin-text-field>
+
+        <vaadin-text-field label="Disabled" placeholder="Placeholder" disabled></vaadin-text-field>
+
+        <vaadin-text-field label="Read-only" placeholder="Placeholder" readonly></vaadin-text-field>
+
+        <vaadin-text-field label="Required" placeholder="I need a value" required></vaadin-text-field>
+
+        <vaadin-text-field label="With error message" value="Invalid" pattern="[0-9]+" invalid error-message="Numbers only!"></vaadin-text-field>
+      </template>
+    </demo-snippet>
+
+    <h3>Dark theme</h3>
+    <demo-snippet>
+      <template>
+        <div theme="dark background space-m">
+          <!-- TODO: how to avoid the need to add the theme attribute here (as :host-context is no longer available)? -->
+
+          <vaadin-text-field theme="dark" label="Label" placeholder="Placeholder"></vaadin-text-field>
+
+          <vaadin-text-field theme="dark" label="Focused" value="Value" focused></vaadin-text-field>
+
+          <vaadin-text-field theme="dark" label="Numbers" value="Invalid" pattern="[0-9]+" invalid error-message="Numbers only!"></vaadin-text-field>
```

Comment:
Some weird glitching in Linux/Chrome:

![recorded](https://cloud.githubusercontent.com/assets/6059356/26300063/b6901514-3ee5-11e7-9bf9-2d79bdc70607.gif)